### ISSUE TITLE
Dropdown doesn't bind to a variable when variable value is loaded in ngOnInit

### ISSUE THREAD

**Issue by faisal-zulfiqar** [primefaces/primeng#7188]
```
[x] bug report
```
**Current behavior**

I've an Edit Purchase Invoice component. It has 2 `<p-dropdowns>`. The first one is for supplier and the second one is for purchase manager. When the component is loaded a variable is initialized in ngOnInit() by calling the backend which responds with the corresponding purchase invoice. The incoming invoice is saved in this.purchaseInvoice. 
Now the first p-dropdown is bound with 'purchaseInvoice.supplier' and I immediately see the predefined value selected in the dropdown as the initial value. But the 2nd dropdown bound as [(ngModel)]="purchaseInvoice.purchasedBy" (similar to supplier) doesn't gets initialized with the value present in the property. 
I've also tried binding it with a variable instead of an interface property but for some strange reason, it doesn't work.
**Expected behavior**

The dropdown should show the initial value as present in purchaseInvoice.purchasedBy

**Minimal reproduction of the problem with instructions**

```
<div *ngIf="!isEdit; else purchasedByLabel">
    <p-dropdown
        *ngIf="!multipleCashAccounts; else purchasedByLabel"
        name="purchasedBy"
        [style]="{ width: '100%' }"
        [autoDisplayFirst]="false"
        [options]="purchaseManagers"
        [(ngModel)]="purchaseInvoice.purchasedBy"
        optionLabel="fullName"
        [filter]="true"
        [showClear]="true"
        required
>
    <ng-template let-manager pTemplate="item">
        <div class="p-grid">
            <div class="p-col-4">
                {{ manager.value?.fullName }}
            </div>
            <div class="p-col-8">
                {{ manager.value?.email }}
             </div>
         </div>
     </ng-template>
     </p-dropdown>
</div>
<ng-template #purchasedByLabel>
    <p>{{ purchaseInvoice.purchasedBy?.fullName }}</p>
</ng-template>
```
**What is the motivation / use case for changing the behavior?**

It's a bug, should initaize the dropdown with the correct value.
**Please tell us about your environment:**

OS: Ubuntu 18.04.1
IDE: Visual Studio Code
Package Manager: NPM

* **Angular version:** 7.1.4

* **PrimeNG version:** 7.0.3

* **Browser:** all

 
* **Language:** TypeScript 3.1.6

* **Node (for AoT issues):** `node --version` =  v10.15.0

**Comment by josh-saunders** [comment]
I am having a similar issue as well. I pulled in the component to work on a fix. I added a key each of the objects I was passing into the component of either `selected: true` for the one I wanted to be default or `selected:false` for all of the others.
In the primeng dropdown component I changed 
`ngOnInit() {
    ngOnInit() {
        this.optionsToDisplay = this.options;
        this.updateSelectedOption(null);
    }
}`
to 
`ngOnInit() {
        this.optionsToDisplay = this.options;
        this.updateSelectedOption(this.options.filter(data => data.selected)[0]);
    }
}`
which filters for the one selected option and passes it on to the updateSelectedOptions method which I changed as well.

`    updateSelectedOption(val: any): void {
        if (this.autoDisplayFirst && !this.placeholder && !this.selectedOption && this.optionsToDisplay && !this.editable) {
            this.selectedOption = val;
        }
        this.selectedOptionUpdated = true;
    }`
This allowed me to selected default options initially. I also removed one of the checks inside the if statement because I needed to allow for null values, but that is my own issue.
